### ISSUE TITLE
SESP should follow semantic versioning again

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/SemanticExtraSpecialProperties#217]
### Setup and configuration

Software | Version
-- | --
MediaWiki | 1.35.11 (5740192)30. Jun. 2023, 17:35
PHP | 7.3.31-1~deb10u4 (apache2handler)
MariaDB | 10.3.39-MariaDB-0+deb10u1
Semantic MediaWiki | 3.2.3
Semantic Extra Special Properties | 3.0.2 and beyond

### Issue
```
[ed586310c821abe45fd8f275] [no req]   ParseError from line 35 of /../w/extensions/SemanticExtraSpecialProperties/src/PropertyAnnotators/UserRightPropertyAnnotator.php: syntax error, unexpected 'PermissionManager' (T_STRING), expecting function (T_FUNCTION) or const (T_CONST)
```
As a result, no annotations are done to a page for 3.0.2, 3.0.3, and 3.04. Once I pin 3.0.1, annotating data to pages work again. I am not using special properties like user groups or rights, meaning it is not an issue triggered by a specific property. At least this is how it appears to me.

### Backtrace
```
Backtrace:
#0 [internal function]: AutoLoader::autoload(string)
#1 /../w/extensions/SemanticExtraSpecialProperties/src/PropertyAnnotators/DispatchingPropertyAnnotator.php(154): spl_autoload_call(string)
#2 /../w/extensions/SemanticExtraSpecialProperties/src/PropertyAnnotators/DispatchingPropertyAnnotator.php(88): SESP\PropertyAnnotators\DispatchingPropertyAnnotator->initDefaultPropertyAnnotators()
#3 /../w/extensions/SemanticExtraSpecialProperties/src/PropertyAnnotators/DispatchingPropertyAnnotator.php(73): SESP\PropertyAnnotators\DispatchingPropertyAnnotator->findPropertyAnnotator(SMW\DIProperty)
#4 /../w/extensions/SemanticExtraSpecialProperties/src/ExtraPropertyAnnotator.php(72): SESP\PropertyAnnotators\DispatchingPropertyAnnotator->addAnnotation(SMW\DIProperty, SMW\SemanticData)
#5 /../w/extensions/SemanticExtraSpecialProperties/src/HookRegistry.php(178): SESP\ExtraPropertyAnnotator->addAnnotation(SMW\SemanticData)
#6 /../w/includes/HookContainer/HookContainer.php(329): SESP\HookRegistry->SESP\{closure}(SMW\SQLStore\SQLStore, SMW\SemanticData)
#7 /../w/includes/HookContainer/HookContainer.php(132): MediaWiki\HookContainer\HookContainer->callLegacyHook(string, array, array, array)
#8 /../w/includes/Hooks.php(137): MediaWiki\HookContainer\HookContainer->run(string, array, array)
#9 /../w/extensions/SemanticMediaWiki/src/Store.php(232): Hooks::run(string, array)
#10 /../w/extensions/SemanticMediaWiki/src/DataUpdater.php(415): SMW\Store->updateData(SMW\SemanticData)
#11 /../w/extensions/SemanticMediaWiki/src/DataUpdater.php(294): SMW\DataUpdater->updateData()
#12 /../w/extensions/SemanticMediaWiki/src/DataUpdater.php(198): SMW\DataUpdater->runUpdate()
#13 /../w/extensions/SemanticMediaWiki/src/ParserData.php(454): SMW\DataUpdater->doUpdate()
#14 /../w/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(303): SMW\ParserData->updateStore()
#15 /../w/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(216): SMW\MediaWiki\Jobs\UpdateJob->updateStore(SMW\ParserData)
#16 /../w/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(137): SMW\MediaWiki\Jobs\UpdateJob->parse_content()
#17 /../w/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateJob.php(94): SMW\MediaWiki\Jobs\UpdateJob->doUpdate()
#18 /../w/extensions/SemanticMediaWiki/src/Maintenance/DistinctEntityDataRebuilder.php(197): SMW\MediaWiki\Jobs\UpdateJob->run()
#19 /../w/extensions/SemanticMediaWiki/src/Maintenance/DistinctEntityDataRebuilder.php(170): SMW\Maintenance\DistinctEntityDataRebuilder->doUpdate(SMW\MediaWiki\JobFactory, Title)
#20 /../w/extensions/SemanticMediaWiki/src/Maintenance/DataRebuilder.php(267): SMW\Maintenance\DistinctEntityDataRebuilder->doRebuild()
#21 /../w/extensions/SemanticMediaWiki/src/Maintenance/DataRebuilder.php(318): SMW\Maintenance\DataRebuilder->rebuildFromSelection(array)
#22 /../w/extensions/SemanticMediaWiki/src/Maintenance/DataRebuilder.php(222): SMW\Maintenance\DataRebuilder->rebuildAll()
#23 /../w/extensions/SemanticMediaWiki/maintenance/rebuildData.php(198): SMW\Maintenance\DataRebuilder->rebuild()
#24 /../w/maintenance/doMaintenance.php(107): SMW\Maintenance\rebuildData->execute()
#25 /../w/extensions/SemanticMediaWiki/maintenance/rebuildData.php(294): require_once(string)
#26 {main}
```
To summarize, version 3.0.2 dropped compatibility with SMW 3.2.x, which should be documented in the release notes. I am not expecting a fix, but the next release should be a 3.1.x or even 4.0.x Perhaps we could do a release now to mitigate the issue.

**Comment by gesinn-it-gea** [comment]
SMW 3.x is still in composer.json but not covered by CI. I doubt that everything will work for SMW 3.x. Is there any reason why you need to stick to SMW 3? Regarding SemVer: I think the (soft) dropping of support for SMW 3 was expressed by the release of 3.x of SESP...

**Comment by kghbln** [comment]
The issue for still staying with SMW 3.x was, so far, the incompatibility of the Semantic Cite extension. If think this was fixed recently. This issue concerns making people still sticking to SMW 3.x aware of it.

Soft dropping is something that semver does not support, if I am not mistaken. Also, from reading the release notes, a regular user cannot conclude that support SMW 3.x was dropped along the way.

Perhaps just expanding the release notes for 3.0.0 (no official SMW 3.x support) may be a solution here.

**Comment by gesinn-it-gea** [comment]
You are absolutely right that the compatibility is obviously not clearly shown. If it causes problems even for you as a power user, then even more so for many others.

**Comment by kghbln** [comment]
The issue here was to detect that annotations do not work, though everyday usage revealed it after a couple of days. After that, pinning 3.0.1 was not a big deal. One needs to know that everything breaks after going beyond 3.0.1
